pkg/webhook/controlplane/ensurer_test.go

```diff
@@ -301,104 +265,10 @@ var _ = Describe("Ensurer", func() {
 				},
 			}
 
-			err := ensurer.EnsureKubeControllerManagerDeployment(ctx, eContextK8s126, dep, nil)
-			Expect(err).To(Not(HaveOccurred()))
-
-			checkKubeControllerManagerDeployment(dep, "1.26.0")
-		})
-	})
-
-	Describe("#EnsureKubeSchedulerDeployment", func() {
-		var (
-			dep     *appsv1.Deployment
-			ensurer genericmutator.Ensurer
-		)
-
-		BeforeEach(func() {
-			dep = &appsv1.Deployment{
-				ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: v1beta1constants.DeploymentNameKubeScheduler},
-				Spec: appsv1.DeploymentSpec{
-					Template: corev1.PodTemplateSpec{
-						Spec: corev1.PodSpec{
-							Containers: []corev1.Container{
-								{
-									Name: "kube-scheduler",
-								},
-							},
-						},
-					},
-				},
-			}
-
-			ensurer = NewEnsurer(logger)
-		})
-
-		It("should add missing elements to kube-scheduler deployment (k8s < 1.26)", func() {
-			err := ensurer.EnsureKubeSchedulerDeployment(ctx, eContextK8s125, dep, nil)
-			Expect(err).To(Not(HaveOccurred()))
-
-			checkKubeSchedulerDeployment(dep, "1.25.0")
-		})
-
-		It("should add missing elements to kube-scheduler deployment (k8s = 1.26)", func() {
-			err := ensurer.EnsureKubeSchedulerDeployment(ctx, eContextK8s126, dep, nil)
-			Expect(err).To(Not(HaveOccurred()))
-
-			checkKubeSchedulerDeployment(dep, "1.26.0")
-		})
-
-		It("should add missing elements to kube-scheduler deployment (k8s >= 1.27)", func() {
-			err := ensurer.EnsureKubeSchedulerDeployment(ctx, eContextK8s127, dep, nil)
-			Expect(err).To(Not(HaveOccurred()))
-
-			checkKubeSchedulerDeployment(dep, "1.27.1")
-		})
-	})
-
-	Describe("#EnsureClusterAutoscalerDeployment", func() {
-		var (
-			dep     *appsv1.Deployment
-			ensurer genericmutator.Ensurer
-		)
-
-		BeforeEach(func() {
-			dep = &appsv1.Deployment{
-				ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: v1beta1constants.DeploymentNameClusterAutoscaler},
-				Spec: appsv1.DeploymentSpec{
-					Template: corev1.PodTemplateSpec{
-						Spec: corev1.PodSpec{
-							Containers: []corev1.Container{
-								{
-									Name: "cluster-autoscaler",
-								},
-							},
-						},
-					},
-				},
-			}
-
-			ensurer = NewEnsurer(logger)
-		})
-
-		It("should add missing elements to cluster-autoscaler deployment (k8s < 1.26))", func() {
-			err := ensurer.EnsureClusterAutoscalerDeployment(ctx, eContextK8s125, dep, nil)
-			Expect(err).To(Not(HaveOccurred()))
-
-			checkClusterAutoscalerDeployment(dep, "1.25.0")
-		})
-
-		It("should add missing elements to cluster-autoscaler deployment (k8s >= 1.26)", func() {
-			err := ensurer.EnsureClusterAutoscalerDeployment(ctx, eContextK8s126, dep, nil)
-			Expect(err).To(Not(HaveOccurred()))
-
-			checkClusterAutoscalerDeployment(dep, "1.26.0")
```

Comment:
You could also remove the `checkClusterAutoscalerDeployment` func. It is no longer used.